prober/http_test.go

```diff
@@ -1806,3 +1807,60 @@ func TestBody(t *testing.T) {
 		}
 	}
 }
+
+func TestValidHTTPVersionsQUIC(t *testing.T) {
+	tests := []struct {
+		ValidHTTPVersions []string
+		ShouldSucceed     bool
+	}{
+		{[]string{"HTTP/1.1", "HTTP/2.0"}, false},
+		{[]string{"HTTP/3.0"}, true},
+		{[]string{"HTTP/1.1"}, false},
+		{[]string{"HTTP/2.0"}, false},
+		{[]string{}, true},
+	}
+	for i, test := range tests {
+		recorder := httptest.NewRecorder()
+		registry := prometheus.NewRegistry()
+		result := ProbeHTTP(context.Background(), "https://http3.is",
```

Comment:
I would like us to NOT depend on a third party website to run our unit tests, I prefer that we don't depend on https://http3.is in our unit tests.

you can setup an simple http3 test server, run it in test and probe that, also we will have a better control over that server and we can remove the dependency on `https://http3.is`

also, dependency on https://http3.is means that unit tests are reaching out on the public internet, and can be ran offline.